helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java

```diff
@@ -117,9 +121,20 @@ public HelixDataAccessor getDataAccssor(String clusterName) {
     }
   }
 
+  public ZkBaseDataAccessor<ZNRecord> getZkBaseDataAccessor(ZkSerializer serializer) {
```

Comment:
Here's an alternative design: What do you think about creating a base data accesor on-demand every time this endpoint gets called? The REST server should be a stateless API server in general anyways, so I don't see anything vastly wrong with that approach. 

And the way we create ZkBaseDataAccessor, we use a SharedZkClient, so we know we aren't creating a ZkConnection every time (we would be re-using the ZkConnection)?

Reply:
It's a workable solution. I took a look at the code, the difficulty is getting `String zkAddress` in accessor class. The current pool also creates only one zkConnection as well, isn't it?

Reply:
Sorry I don't understand your comment. Could you please clarify?

Reply:
For your idea, the propertyStoreAccessor needs to `new ZkBaseDataAccessor(_zkAddress)`
every time, the `_zkAddress` is a private field. We need to understand deeper that the ZkBaseDataAccessor constructor uses the same ZkConnection to realize that we're not creating multiple ZkConnections. We'd better decouple the logic and this kind of pre-knowledge. 

My implementation of the proposed "zkBaseDataAccessor" pool also creates only one instance of zkConnection for propertyStoreAccessor as well, isn't it?

Reply:
> We need to understand deeper that the ZkBaseDataAccessor constructor uses the same ZkConnection to realize that we're not creating multiple ZkConnections. We'd better decouple the logic and this kind of pre-knowledge.

You could figure this out by reading how SharedZkClient is implemented. It shares one ZkConnection.

> My implementation of the proposed "zkBaseDataAccessor" pool also creates only one instance of zkConnection for propertyStoreAccessor as well, isn't it?

I believe your current implementation creates a ZkBaseDataAccessor instance per serializer. In a pathological case, I call your read method with 20K serializers, then you would have 20K different ZkBaseDataAccessors?

Reply:
My current implementation only creates one instance for all operations related to propertyStore, it's verified in the unit test.